AdobeStockImage/Model/GetImageByAdobeId.php

```diff
@@ -0,0 +1,90 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+declare(strict_types=1);
+
+namespace Magento\AdobeStockImage\Model;
+
+use Magento\AdobeStockAsset\Model\DocumentToAsset;
+use Magento\AdobeStockAssetApi\Api\Data\AssetInterface;
+use Magento\AdobeStockImageApi\Api\GetImageListInterface;
+use Magento\Framework\Api\FilterBuilder;
+use Magento\Framework\Api\Search\SearchCriteriaBuilder;
+use Magento\Framework\Exception\NotFoundException;
+
+/**
+ * Class GetImageByAdobeId
+ *
+ * Service for getting image by content id
+ */
+class GetImageByAdobeId
```

Comment:
This class implementation is not specific to images. It can be moved to AdobeStockAsset module and renamed to `GetAssetById` (adobe_id equals to internal id)

In this case `Magento\AdobeStockAssetApi\Api\GetAssetListInterface`  can be used insgead of `GetImageListInterface`